src/esp/assets/BaseMesh.cpp

```diff
@@ -19,5 +24,102 @@ bool BaseMesh::setMeshType(SupportedMeshType type) {
+    std::vector<std::string>& debugMsgs) const {
+  // build per-SSD object vector of known semantic IDs
+  std::size_t numSSDObjs = ssdObjs.size();
+  // no semantic ID 0
+  std::vector<int> semanticIDToSSOBJidx(numSSDObjs + 1);
+  for (int i = 0; i < numSSDObjs; ++i) {
+    const auto& ssdObj = *ssdObjs[i];
+    int semanticID = ssdObj.semanticID();
+    if (semanticIDToSSOBJidx.size() <= semanticID) {
+      semanticIDToSSOBJidx.resize(semanticID + 1);
+    }
+    semanticIDToSSOBJidx[semanticID] = i;
+  }
+
+  // aggegates of per-semantic ID mins and maxes
+  std::vector<esp::vec3f> vertMax(
+      semanticIDToSSOBJidx.size(),
+      {-Mn::Constants::inf(), -Mn::Constants::inf(), -Mn::Constants::inf()});
+  std::vector<esp::vec3f> vertMin(
+      semanticIDToSSOBJidx.size(),
+      {Mn::Constants::inf(), Mn::Constants::inf(), Mn::Constants::inf()});
+  std::vector<int> vertCounts(semanticIDToSSOBJidx.size());
+
+  // for each vertex, map vert min and max for each known semantic ID
+  // Known semantic IDs are expected to be contiguous, and correspond to the
+  // number of unique ssdObjs mappings.
+
+  for (int vertIdx = 0; vertIdx < vertSemanticIDs.size(); ++vertIdx) {
+    // semantic ID on vertex - valid values are 1->semanticIDToSSOBJidx.size().
+    // Invalid/unknown semantic ids are > semanticIDToSSOBJidx.size()
+    const auto semanticID = vertSemanticIDs[vertIdx];
+    if ((semanticID > 0) && (semanticID <= semanticIDToSSOBJidx.size())) {
+      const auto vert = vertices[vertIdx];
+      // FOR VERT-BASED OBB CALC
+      // only support bbs for known colors that map to semantic objects
+      vertMax[semanticID] = vertMax[semanticID].cwiseMax(vert);
+      vertMin[semanticID] = vertMin[semanticID].cwiseMin(vert);
+      vertCounts[semanticID] += 1;
+    }
+  }
+
+  // with mins/maxs per ID, map to objs
+  // give each ssdObj the values to build its OBB
+  debugMsgs.reserve(semanticIDToSSOBJidx.size());
+  for (int semanticID = 1; semanticID < semanticIDToSSOBJidx.size();
+       ++semanticID) {
+    // get object with given semantic ID
+    auto& ssdObj = *ssdObjs[semanticIDToSSOBJidx[semanticID]];
+    esp::vec3f center{};
+    esp::vec3f dims{};
+    const std::string debugStr = Cr::Utility::formatString(
+        "Semantic ID : {} : color : {} tag : {} present in {} verts | ",
+        semanticID, ssdObj.getColorAsString(), ssdObj.id(),
+        vertCounts[semanticID]);
+    std::string infoStr;
+    if (vertCounts[semanticID] == 0) {
+      infoStr = Cr::Utility::formatString(
+          "{}No verts have specified Semantic ID.", debugStr);
+    } else {
+      center = .5f * (vertMax[semanticID] + vertMin[semanticID]);
+      dims = vertMax[semanticID] - vertMin[semanticID];
+      infoStr = Cr::Utility::formatString(
+          "{}BB Center [{},{},{}] Dims [{},{},{}]", debugStr, center.x(),
+          center.y(), center.z(), dims.x(), dims.y(), dims.z());
+    }
+    debugMsgs.emplace_back(std::move(infoStr));
```

Comment:
I'm *very* not fond of populating a `std::vector<std::string>` with debug messages that get ignored / thrown away 99.98% of the time. 

What we really invested time into with Erik Wijmans during the logging refactor was making `ESP_DEBUG()` and other macros an overhead-free no-op if given log output is disabled -- the actual debug string is not even built in that case. So please use them, even if you'd have to do some extra work filtering or redirecting the output.